org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/InstanceOfExpression.java

```diff
@@ -385,6 +385,14 @@ public void traverse(ASTVisitor visitor, BlockScope scope) {
 			this.type.traverse(visitor, scope);
 		}
 	}
+	if (this.pattern instanceof RecordPattern) {
```

Comment:
After the recent changes, I think it is possible to simply delegate to the this.pattern regardless of the pattern type. That way we can get rid of the elementVariable.traverse() call too.

Reply:
Thanks ! I updated accordingly. Looks much cleaner.